pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -279,21 +283,31 @@ func (c *Controller) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthE
 		return
 	}
 
-	// get previous status
-	cfg := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		log.Errorf("config was nil when getting WorkloadEntry %v for %v", entryName, proxy.ID)
-		return
-	}
+	condition := transformHealthEvent(event)
+	// backoff retry
+	// TODO(@hzxuzhonghu): handle race when reconnect happens and health status changed
+	for i := 0; i < maxRetries; i++ {
+		if i > 0 {
+			time.Sleep(waitTime << (i - 1))
```

Comment:
nit: can we use decimal math instead of bit math? Its a lot easier to grok for most